File: compare/parse.pl
use warnings FATAL => 'all';
use strict;
use utf8;
use Data::Dumper::Simple;
use Perl6::Say;

binmode(STDIN,':utf8');
binmode(STDOUT,':utf8');

my $start = localtime();

my $filename = '21103114.log';
open(my $fh, '<:encoding(UTF-8)', $filename)
    or die "Could not open file '$filename' $!";

my @arr;
my $str_log = "";
my @arrtj;

while (my $str = <$fh>) {
    chomp $str;

    if ($str =~ /([0-9]{2}):([0-9]{2})\.([0-9]+)\-([0-9]+)\,(\w+)\,(\d+)/) {
        if (length($str_log)) {
            push(@arr, $str_log);
        }
        $str_log = $str;
    }
    else {
        $str_log = join("-#-", $str_log, $str);
    }
}
say scalar @arr;

for (my $i=0; $i<scalar @arr; $i++) {
    my %tj;
    while ($arr[$i] =~ m/,([A-Za-z0-9_А-Яа-я:]+)=([^,]+)/g) {
        $tj{$1} = $2;
    }
    push(@arrtj, \%tj);
}
say scalar @arrtj;
#warn Dumper @arrtj;

say localtime();
my $duration = localtime() - $start;
say ($duration);

File: compare/parse.py
import re
from datetime import datetime
import time

#filename = '../21103114.log'
filename = '../21121413.log'#big

start_time = datetime.now()

str_log = ""
arr = list()

with open(filename, "r", encoding="utf-8") as f:
    for str in f.readlines():
        str.strip()

        if (re.match(r'([0-9]{2}):([0-9]{2})\.([0-9]+)\-([0-9]+)\,(\w+)\,(\d+)', str)):
            if (len(str_log)):
                arr.append(str_log)

            str_log = str

        else:
            str_log = str_log + str

print("len array: ")
print(len(arr))

for elem in arr:
    param = re.findall(',([A-Za-z0-9_А-Яа-я:]+)=([^,]+)', elem)
    len(param)

stop_time = datetime.now()
print("start: ")
print(start_time)

print("stop: ")
print(stop_time)
print("diff: ")
print(stop_time - start_time)